Bump egui version to 0.14.2

Ragarnoy opened the pull request:
I made sure it would compile, and fixed a bug that made the crate unusable, but I couldn't find a fix to the regression of the address bar, I can't seem to be able to detect focus on the singlelign widget, could be a bug introduced by 0.14. I will update the PR if i manage to fix it.

Edit: I think I fixed it ! (?)

Ragarnoy committed:
Merge pull request #1 from Ragarnoy/feature/update_egui

Update to 0.13.1, fix deprecated functions and update click sensing o…

Ragarnoy committed:
Merge branch 'Hirtol:master' into master

Ragarnoy committed:
Add condition to prevent underflow when pressing up arrow key if user is on first line of the memory editor.

Ragarnoy committed:
Code cleanup

Ragarnoy committed:
Add option field to set whether the column number is modifiable or not.

Ragarnoy committed:
Add option field to set whether `Options` is collapsed by default or not.

Hirtol committed:
Small documentation changes for the new options.

Ragarnoy committed:
Merge branch 'Hirtol:master' into master

Ragarnoy committed:
bump egui and eframe to 0.14.2

Ragarnoy committed:
.on_hover_text() now requires .response

Ragarnoy committed:
Fixed unwanted behaviour: Without setting a desired width to the singleline widget, attempting to edit a memory variable would resize all the columns in a weird way.

Ragarnoy committed:
add newline to end of file

Ragarnoy committed:
increase readability a bit

Ragarnoy committed:
response.clicked() was fixed, I changed the response to lost_focus, however the check needed to occur after checking if enter was pressed, otherwise you would clear goto_address_string before scrolling to the line, causing a parse error.

Hirtol commented:
Thanks for looking at this again!

I tried updating this earlier in the week, but encountered an issue which this PR suffers from as well at the moment. If you try to move to `FF50` via the goto option in the example you'll find that the scroll isn't correctly offset (and `FF50` isn't in view). This only seem to matter for things greater than address `FF00`, so I presume it's only relevant for when you're close to the edge of the selected address space.

Lastly, your last commit changes the behaviour of the `goto` function slightly. Sadly, the `response.clicked()` still isn't 'fixed' (in that it works the way it used to work), as it's still triggered by pressing 'enter'. This means that the field would indeed be cleared before we execute the goto if you remove the `enter` check. Therefore the check is necessary. Currently the goto field isn't cleared when we click on the field due to this change, and I'd personally like to keep the old behaviour instead of clearing on loss of focus.

To keep the old behaviour all that needs to be changed compared to the previous version is: `response.has_focus()` -> `response.lost_focus()`.

If you're alright with it I'll push that change (and hopefully a fix for the `FF50` goto issue) tomorrow.

Hirtol committed:
Revert the goto option field changes, and update it to work with the new egui version.

Hirtol committed:
'Fix' the goto scroll issue caused by the version upgrade.

The core of the issue seems to be that Egui tries to align a row to the top of the scroll area (as ordered), but runs out of elements to pad the bottom with, and thus kind of gives up. Note, this only happens if we're at the end of the address space (as then we no longer have new values to pad the bottom with).

This is a new issue in Egui 0.14, as previous versions made a best effort attempt to still scroll to the desired row, even if they couldn't align it to the top of the screen.

By now instead aligning to the Center of the screen it once again seems to do a best effort scroll, and scroll to the row we want, even if it can't quite align it at the centre of the screen.

Hirtol committed:
Fix unchecked subtract
